data/tutorials/ds_02_set.md

```diff
@@ -7,140 +7,253 @@ category: "data-structures"
+remove an element from your set, and so on).
+
+Note also that this module defines two types: `type elt = String.t`, representing
+the type of the elements, and `type t = Set.Make(String).t`, representing the type of
+the set itself. It's important to note this because these types are used in the
+signatures of many of the functions defined in this module.
+
+For example, the `add` function has the signature `elt -> t -> t`, which means
+that it expects an element (a String), and a set of strings, and will return to you
+a set of strings. As you gain more experience in OCaml and other functional languages,
+the type signature of functions are often a basic but very convenient form of documentation
+on how to use those functions.
+
+## Creating a Set
+
+You've created your module representing a set of strings, but now you actually want
+to create an instance of a set of strings. So how do we go about doing this? Well, you
+could search through the documentation for the original `Set` functor to try and
+find what function or value you should use to do this, but this is an excellent
+opportunity to practice reading the type signatures and inferring the answer from them.
+
+You want to create a new set (as opposed to modifying an existing set). So you should
+look for functions whose return result has type `t` (the type representing the set)
+and which *does not* require a parameter of type `t`.
 
-```ocaml
-# let s = SS.singleton "hello";;
-val s : SS.t = <abstr>
+Skimming through the list of functions in the module, there's only a handful of functions
+that match that criteria: `empty: t`, `singleton : elt -> t`, `of_list : elt list -> t`
+and `of_seq : elt Seq.t -> t`.
+
+Perhaps you already know how to work with lists and sequences in OCaml or
+perhaps you don't. For now, let's assume you don't know, so we'll focus
+our attention on the first two functions in that list: `empty` and `singleton`.
+
+The type signature for `empty` says that it simply returns `t`, i.e., an instance
+of our set, without requiring any parameters at all. By intuition, you might
+guess that the only reasonable set that a library function could return when
+given zero parameters is the empty set. And the fact that the function is named
+`empty` reinforces this theory.
+
+Is there a way to test this theory? Perhaps if we had a function that
+could print out the size of a set, then we could check if the set we get
+from `empty` has a size of zero. In other words, we want a function which
+receives a set as a parameter and returns an integer as a result. Again,
+skimming through the list of functions in the module, we see there is a
+function which matches this signature: `cardinal : t -> int`. If you're
+not familiar with the word "cardinal," you can look it up on Wikipedia
+and notice that it basically refers to the size of sets, so this reinforces
+the idea that this is exactly the function we want.
```

Comment:
I would just give the definition here without alluding to wikipedia, to make it more self-contained. Same for singletons and Purely Functional Data Structure.

It gives a very unformal tone to this page, which is also the case for "Data structures > Map", but not for the others. Related to #1154.